Framework/Algorithms/src/BinaryOperation.cpp

```diff
@@ -760,8 +752,6 @@ void BinaryOperation::do2D(bool mismatchedSpectra) {
       int64_t rhs_wi = i;
       if (mismatchedSpectra && table) {
         rhs_wi = (*table)[i];
-        if (rhs_wi < 0)
-          continue;
```

Comment:
For which purpose you modify BinaryOperation (in general)?

Earlier in this code, the behavior of the rhs_wi < 0 case is treated differently. Thus, I guess it should be everywhere like you propose or not being changed?

Reply:
It changes the operation for an edge case when the first term (dividend) has more spectra than the second one (divisor). The current behaviour is to skip these spectra which results in zeroing them. I change that to copying such spectra from the input unchanged. I need it for flood correction to work in RROA: 

The correction workspace doesn't include monitors but the data and transmission ones do. The flood ws divides both and we need to keep the monitor spectra preserved for the subsequent normalisation.

Reply:
It's somehow smart to avoid something like `ExtractMonitorWorkspace` (as we do in Python).

This changes the behavior of `Divide` (for histograms only?) and needs documentation then?

Reply:
And the documentation of all other algorithms which use BinaryOperator if they do not do some compatibility checks (Maybe `Divide`, `Multiply`, `WeightedMean`, `Plus`)?